### ISSUE TITLE
Instant crash to DeltaTouch home screen when using Gun Bonsai with Brutal Doom Platinum

### ISSUE THREAD

**Issue by pbates5** [ToxicFrog/doom-mods#11]
I'm using Gun Bonsai 0.9.4, libtooltipmenu 0.2.1 with the latest ver of Brutal Doom Platinum 3.1.1 on GZDoom v4.8.2. I used to be able to play previous versions of these mods together with no issues but clearly something is conflicting now because I'm getting consistent crashes on multiple wads (Doom 2, Freedoom, Evilution). The crashes usually don't start until about;5 or 6 maps in, when I've had enough time to accumulate upgrades for my weapons. My mod order is as follows:

Brutal Doom Platinum v3.1.1
libtooltipmenu v0.2.1
Gun Bonsai 0.9.4
bullet-time-x
morelights v2.4
Universal Rain and Snow

I tested this mod order first with Gun Bonsai turned off then Brutal Doom Platinum turned off and they seemed to run fine without a single crash on the first several maps, so I think it's specifically a conflict between BDP and GB. Unfortunately I don't even get a crash log when it happens, after about a couple minutes of gameplay no matter where I am or what's happening, my game gives a slight stutter then crashes to the DeltaTouch home screen, so I'm not sure if I can provide more specific information.

My hardware specs:
Android ver 12
Qualcomm Snapdragon 888 3 GHz
18 GB RAM

**Comment by ToxicFrog** [comment]
I can't test it in DeltaTouch but I'll give it a look in amd64 gzdoom and see if I can reproduce -- if I can I should at least be able to get a stack trace.

**Comment by pbates5** [comment]
Thank you I appreciate your dedication to this amazing mod. It's the most fun I've had with Doom in decades 🥰

**Comment by ToxicFrog** [comment]
Ok, I could not reproduce this with the following setup:
- gzDoom 4.10.0, linux amd64
- libtooltipmenu 0.2.3
- Gun Bonsai 0.9.6
- Brutal Doom Platinum 3.1.1
- Bullet Time X 111.1
- More Lights 2.5
- Universal Rain and Snow

I played through the first few levels of doom 2 without issues, then woke up a bunch of monsters and let it sit there with them hammering on me in god mode while I did other stuff for half an hour, and it ran fine. 

The most optimistic possibility is that this means you just need to update something to a more recent version and things will be fixed. I think updating gzDoom from 4.8 to 4.10 is most likely to help, since an immediate crash with no vmabort message is *usually* a weird engine bug rather than a weird zscript bug, but it's possible updating GB will result in changes that cause the bug to no longer trigger  — 0.9.5, in particular, included a bunch of internal structural changes to better support multiplayer.

If that doesn't help, it's probably either a Delta Touch specific bug — in which case I'm not going to be able to reproduce it on desktop, and there's not much I can do — or there's something different about how I play that means I don't trigger the bug even if you do. To help investigate the latter case, it might be useful to do more testing -- say, save frequently, and if it crashes reload your most recent save and see if it crashes again in the same place. It would help to know if the crash is truly random, or if there's some specific thing that triggers it.

**Comment by pbates5** [comment]
I updated GZDoom to v4.10, GB to 0.9.5, and libttm to 0.2.2. I started a new game on each .wad (unfortunately I lost all my old save files after update) and continued playing with my old mod order. Everything seemed to be working fine on Doom, Doom 2, and Freedom for the first few maps. That is until I started Evilution and experienced the same crash to home screen on the second map (specifically in the first hallway right when the archvile becomes activated in the outdoor area on the left). I turned off all my mods except BDP, GB, and libttm and crashed in the exact same area.

**Comment by ToxicFrog** [comment]
I'm leaving this open because it still doesn't seem to be fixed, but since I can't reproduce it on desktop and the crash in DeltaTouch doesn't leave a stack trace, there isn't much I can do. :( Please report back if you get more info (or if it stops happening), though.

**Comment by pbates5** [comment]
I apologize I haven't commented in so long, I've rarely had time to play delta touch in the past several months. However, the few times I have I've experienced no crashes since GB v10 (or whichever the version that merged libttm with the main mod).

Unfortunately there now seems to be a new issue with the latest version of GZ Doom (v4.11.3). I'll open a new thread so this one may be closed.
